lib/datasource/artifactory/index.ts

```diff
@@ -0,0 +1,97 @@
+import { logger } from '../../logger';
+import { ExternalHostError } from '../../types/errors/external-host-error';
+import { cache } from '../../util/cache/package/decorator';
+import { parse } from '../../util/html';
+import { HttpError } from '../../util/http/types';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import { datasource, defaultRegistryUrl } from './common';
+
+export class ArtifactoryDatasource extends Datasource {
+  static readonly id = datasource;
+
+  constructor() {
+    super(datasource);
+  }
+
+  override readonly customRegistrySupport = true;
+
+  override readonly defaultRegistryUrls = [defaultRegistryUrl];
```

Comment:
To be changed

Reply:
```suggestion
```

Not remove, if there is no default.

Reply:
Finally removed as suggested in comments below.